packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.js

```diff
@@ -87,26 +124,111 @@ export default class QuanticSearchBoxSuggestionsList extends LightningElement {
     }
   }
 
-  get suggestionsToRender() {
-    return this.suggestions?.map((s, i) => ({
-      ...s,
-      isSelected: this.selectionIndex === i,
-    }));
+  get allOptions() {
+    if (this.shouldDisplayRecentQueries) {
+      const options = [
+        ...this.filteredRecentQueries,
+        ...this.filteredSuggestions,
+      ]?.map((option, index) => ({
+        ...option,
+        key: index + 1,
+        isSelected: this.selectionIndex === index + 1,
+        containerCSSClass: `${optionCSSClass} ${this.selectionIndex === index + 1 ? 'slds-has-focus' : ''}`,
+        icon: option.isRecentQuery ? 'utility:clock' : 'utility:search',
+        clickHandler: () => {
+          this.handleSuggestionSelection(option.rawValue, option.isRecentQuery);
+        },
+      }));
+      return [
+        {
+          key: 0,
+          isSelected: this.selectionIndex === 0,
+          isClearRecentQueryButton: true,
```

Comment:
Mainly because the Quantic Search Box Suggestions List has the logic to control what suggestion to highlight when clicking the up and down arrow of the keyboard. also to select a suggestion using the keyboard. and all this rely on indexes.

so if we put  "clearRecentQueries" buttons outside the `template:foreach` we will be then obliged to re-work this logic that handles navigating and highlighting the different options when using the keyboard,  and it will be kinda hard to deal with the indexes as all the suggestions will be indexed from 0 to n  but the  "clearRecentQueries" buttons are not being considered in this so we will need to add conditions like the following:

- if index === n and down arrow key is selected, highlight  "clearRecentQueries" buttons and not the suggestion at index 0.
- if index === 0 and up arrow key is selected, highlight  "clearRecentQueries" buttons and not the suggestion at index n.

For that reason and to avoid that, I saw it will be better to include  "clearRecentQueries" buttons in the forEach so all the logic based on the index will continue to work naturally.

Reply:
Ahh yea that makes sense, it does make the code more complex though :(

Reply:
Do we REALLY need to have the clear recent queries as part of that though?

Could we perhaps have 

```
searchbox [down]
suggestions [up/down]
clear button [down] <- but displayed as the first element because of flex-ordering for example?
```

As in always have that one at the end instead of being the first one? Would that simplify things?